src/main/kotlin/com/terraformation/backend/support/atlassian/AtlassianHttpClient.kt

```diff
@@ -0,0 +1,110 @@
+package com.terraformation.backend.support.atlassian
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.customer.model.requirePermissions
+import com.terraformation.backend.support.atlassian.model.ServiceDeskModel
+import com.terraformation.backend.support.atlassian.model.ServiceRequestFieldsModel
+import com.terraformation.backend.support.atlassian.model.ServiceRequestTypeModel
+import com.terraformation.backend.support.atlassian.resource.AtlassianResource
+import com.terraformation.backend.support.atlassian.resource.CreateServiceDeskRequest
+import com.terraformation.backend.support.atlassian.resource.DeleteIssue
+import com.terraformation.backend.support.atlassian.resource.ListServiceDesks
+import com.terraformation.backend.support.atlassian.resource.ListServiceRequestTypes
+import com.terraformation.backend.support.atlassian.resource.PostServiceDeskRequestResponse
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.java.Java
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.defaultRequest
+import io.ktor.client.request.accept
+import io.ktor.client.request.request
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.serialization.jackson.JacksonConverter
+import jakarta.inject.Named
+import kotlinx.coroutines.runBlocking
+
+/** Submits API requests to interact with Atlassian services. */
+@Named
+class AtlassianHttpClient(private val config: TerrawareServerConfig) {
+  private val httpClient: HttpClient by lazy { createHttpClient() }
+  private val serviceDesk: ServiceDeskModel? by lazy { findServiceDesk() }
+
+  val requestTypes: Set<ServiceRequestTypeModel> by lazy { getServiceRequestTypes() }
+
+  fun deleteIssue(issueId: String) {
+    requirePermissions { deleteSupportIssue() }
+    makeRequest(DeleteIssue(issueId))
+  }
+
+  fun createServiceDeskRequest(
+      description: String,
+      summary: String,
+      requestTypeId: Int,
+      reporter: String,
+  ): PostServiceDeskRequestResponse {
+    // No required permission
+    if (requestTypes.find { it.id == requestTypeId } == null) {
+      throw RuntimeException("Request ID type not recognized")
+    }
+    return makeRequest(
+        CreateServiceDeskRequest(
+            reporter = reporter,
+            requestFieldValues = ServiceRequestFieldsModel(summary, description),
+            requestTypeId = requestTypeId,
+            serviceDeskId = serviceDeskId(),
+        ))
+  }
+
+  private fun serviceDeskId(): Int =
+      if (serviceDesk == null) {
+        throw IllegalStateException("Atlassian configuration is invalid")
+      } else {
+        serviceDesk!!.id
+      }
```

Comment:
I am hoping that this will cause an error during a service call time, instead of during server initialization time. I don't want our whole server to be not running because the Atlassian API is unreachable for example. 

Perhaps I am overthinking it and not understanding the Spring @named components lifecycle well.

Reply:
The `by lazy` is what stops that from happening. The value won't be initialized until it's referenced, which won't happen until something calls into the class. Net effect should be the same as what's in the PR right now: you get an exception at runtime when you call one of the methods that makes a request.

Reply:
Thanks!!